internal/users/locking/testutils.go

```diff
@@ -17,28 +35,163 @@ var overrideLocked atomic.Bool
 func Z_ForTests_OverrideLocking() {
 	testsdetection.MustBeTesting()
 
+	overriddenMu.Lock()
+	defer overriddenMu.Unlock()
+
+	overriddenWriteLockImpl = append(overriddenWriteLockImpl, writeLockImpl)
 	writeLockImpl = func() error {
 		if !overrideLocked.CompareAndSwap(false, true) {
-			return ErrLock
+			return fmt.Errorf("%w: already locked!", ErrLock)
+		}
+
+		log.Debug(context.Background(), "TestOverride: Local entries locked!")
+		return nil
+	}
+
+	overriddenWriteUnlockImpl = append(overriddenWriteUnlockImpl, writeUnlockImpl)
+	writeUnlockImpl = func() error {
+		if !overrideLocked.CompareAndSwap(true, false) {
+			return fmt.Errorf("%w: already unlocked!", ErrUnlock)
+		}
+
+		log.Debug(context.Background(), "TestOverride: Local entries unlocked!")
+		return nil
+	}
+}
+
+// Z_ForTests_OverrideLockingWithCleanup is a function to override the locking
+// functions for testing purposes.
+// It simulates the real behavior but without actual file locking.
+// This implicitly calls [Z_ForTests_RestoreLocking] once the test is completed.
+//
+// nolint:revive,nolintlint // We want to use underscores in the function name here.
+func Z_ForTests_OverrideLockingWithCleanup(t *testing.T) {
+	t.Helper()
+
+	testsdetection.MustBeTesting()
+
+	Z_ForTests_OverrideLocking()
+	t.Cleanup(Z_ForTests_RestoreLocking)
+}
+
+// Z_ForTests_OverrideLockingAsLockedExternally is a function to
+// override the locking functions for testing purposes.
+// It simulates the real behavior but without actual file locking.
+// This locks the database when called, the lock can be both removed using
+// [WriteUnlock] or cancelling the context. It also implicitly calls
+// [Z_ForTests_RestoreLocking] once the test is completed.
+//
+// nolint:revive,nolintlint // We want to use underscores in the function name here.
+func Z_ForTests_OverrideLockingAsLockedExternally(t *testing.T, ctx context.Context) {
+	t.Helper()
+
+	testsdetection.MustBeTesting()
+
+	overriddenMu.Lock()
+	defer overriddenMu.Unlock()
+
+	t.Cleanup(Z_ForTests_RestoreLocking)
+
+	lockCtx, lockCancel := context.WithCancel(context.Background())
+	lockCancel()
```

Comment:
I suspect you don't actually want to cancel the context here

```suggestion
	defer lockCancel()
```

Reply:
Yes we want to.

I though I had put a comment on it, but the reason is that we want to start the locking with a "blocking" lock, so the context is cancelled by default, but I can explain it better inline

Reply:
Btw we could also defer the cancel, but it would't change anything